src/Functions/array/arrayAUC.cpp

```diff
@@ -249,52 +288,132 @@ class FunctionArrayAUC : public IFunction
+                "Number of arguments for function {} doesn't match: passed {}, should be {}",
+                getName(),
+                number_of_arguments,
+                is_pr ? "2 or 3" : "2, 3 or 4");
+
+        /// `arr_scores` argument validation
+        const DataTypeArray * array_type_scores = checkAndGetDataType<DataTypeArray>(arguments[0].type.get());
+        if (!array_type_scores)
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument (arr_scores) for function {} must be of type Array", getName());
+
+        const auto & nested_type_scores = array_type_scores->getNestedType();
+        if (!isNativeNumber(nested_type_scores))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "First argument (arr_scores) for function {} cannot process values of type {}, should be Number",
+                getName(),
+                nested_type_scores->getName());
+
+        /// `arr_labels` argument validation
+        const DataTypeArray * array_type_labels = checkAndGetDataType<DataTypeArray>(arguments[1].type.get());
+        if (!array_type_labels)
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument (arr_labels) for function {} must be of type Array", getName());
+
+        const auto & nested_type_labels = array_type_labels->getNestedType();
+        if (!isNativeNumber(nested_type_labels) && !isEnum(nested_type_labels))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Second argument (arr_labels) for function {} cannot process values of type {}, should be Number or Enum",
+                getName(),
+                nested_type_labels->getName());
+
+        /// `scale` argument validation
+        bool is_third_arg_scale = !is_pr;
+
+        if (!is_pr && number_of_arguments >= 3)
         {
-            const DataTypeArray * array_type = checkAndGetDataType<DataTypeArray>(arguments[i].type.get());
-            if (!array_type)
-                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The two first arguments for function {} must be of type Array.", getName());
-
-            const auto & nested_type = array_type->getNestedType();
-            if (!isNativeNumber(nested_type) && !isEnum(nested_type))
-                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "{} cannot process values of type {}", getName(), nested_type->getName());
+            if (!isValidBooleanColumn(arguments[2]))
+            {
+                if (number_of_arguments == 4)
+                    throw Exception(
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                        "Third argument (scale) for function {} must be of type const Bool",
+                        getName());
+                /// Postpone the exception in case the third argument can be arr_partial_offsets
```

Comment:
The comment looks like it belongs to the the `if` case but I think it really belongs to the `else` case.